### PR TITLE
Homework for lesson5

### PR DESCRIPTION
**Pull request by dimloko**

### CODE REVIEW

File: fifth-lesson/src/MetodTask18.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/*Найти в строке не только запятые, но и другие знаки препинания. Подсчитать
+общее их количество.*/
+
+
+public class MetodTask18 {
+    public static void main(String[] args) {
+        String str = " Строковый литерал — это последовательность символов, " +
+                "заключенных в двойные кавычки.&? "; // Исходная строка
+        int n = 0; //Счётчик символов
+        char[] symbolsCheck = {',', '.', '-', '?', '!', '|', '%', '/', '=', '_', '&'}; //Массив символов для поиска
```
Comment (TheFallen058) [inline fifth-lesson/src/MetodTask18.java:10]: more efficient will be check char, number and space symbol - everything else will be what you are looking for

File: fifth-lesson/src/MetodTask21.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        time = System.nanoTime() - time;
+        System.out.printf("Сложение методом append класса StringBuilder:\t%6.3f ms\n", time / 1000000.0);
+
+        //append класса StringBuffer
+        time = System.nanoTime();
+        for (int i = 0; i < n; i++) {
+            str3.append("test");
+        }
+        time = System.nanoTime() - time;
+        System.out.printf("Сложение методом append класса StringBuffer:\t%6.3f ms\n", time / 1000000.0);
```
Comment (TheFallen058) [inline fifth-lesson/src/MetodTask21.java:35]: Please run this logic at least 3 times (10 is better) and analyze what is more efficient

File: fifth-lesson/src/MetodTask22.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+записаных по правилам Java, с помощью регулярных выражений и вывести их на
+страницу.*/
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class MetodTask22 {
+    public static void main(String[] args) {
+        String str = "Red-0xFF0000, Yellow - 0xFFFF00, DarkGoldenRod - 0xB8860B";  //Исходная строка
+        Pattern pat = Pattern.compile("[0]{1}[xX]{1}[0-9[a-fA-F]]{6}"); //Компиляция представления регулярного выражения
```
Comment (TheFallen058) [inline fifth-lesson/src/MetodTask22.java]: didn't check all cases but "0x[A-Fa-f0-9]{1,4}" looks more efficient

**Review by TheFallen058 — APPROVED** [review]
Good work, please check my comments

File: fifth-lesson/src/MetodTask18.java (hunk 1 above)
Comment (dimloko) [reply fifth-lesson/src/MetodTask18.java:10]: I don't quite understand how in this example to check for char or number

Comment (TheFallen058) [reply fifth-lesson/src/MetodTask18.java:10]: One way is to use util methods Character.isDigit and Character.isLetter. Another - regexp.

**Pull request merged**